### ISSUE TITLE
Failed to obtain link from URI

### ISSUE THREAD

**Issue by andlessa** [jrnewell/spotify-ripper#164]
Hi,

Everytime I try to download a track/playlist/album I get the error:

>File "/usr/local/lib/python2.7/dist-packages/pyspotify-2.0.5-py2.7-linux-x86_64.egg/spotify/link.py", line 73, in __init__
>    'Failed to get link from Spotify URI: %r' % uri)
>ValueError: Failed to get link from Spotify URI: 'uri'

Any idea of what could be wrong?
The URI I am using is something like:

spotify:track:26xjIZq1zOMza3...

**Comment by jrnewell** [comment]
Yeah that is an invalid Spotify track id  (You can verify it here: https://api.spotify.com/v1/albums/26xjIZq1zOMza3/tracks).

You can do a query search for your track or use the Desktop app to get the correct URI (right-click, "Copy Spotify URI").

**Comment by jrnewell** [comment]
Oh sorry, I didn't realize you only posted the partial id.  If you copied it from the desktop app, it should be correct.

**Comment by andlessa** [comment]
When I ran spotify-ripper I did use the full ID. Here is an example with a full ID copied from spotify-desktop:

> spotify-ripper uri spotify:track:26xjIZq1zOMza3mfemA4Q8 -u andlessa -p <XXX>

and I get:

```
Logging in...
Logged in as andlessa
Exception in thread SpotifyRipperThread:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/local/lib/python2.7/dist-packages/spotify_ripper-2.7.2-py2.7.egg/spotify_ripper/ripper.py", line 217, in run
    tracks = get_tracks_from_uri(uri)
  File "/usr/local/lib/python2.7/dist-packages/spotify_ripper-2.7.2-py2.7.egg/spotify_ripper/ripper.py", line 212, in get_tracks_from_uri
    return self.load_link(uri)
  File "/usr/local/lib/python2.7/dist-packages/spotify_ripper-2.7.2-py2.7.egg/spotify_ripper/ripper.py", line 368, in load_link
    link = self.session.get_link(uri)
  File "/usr/local/lib/python2.7/dist-packages/pyspotify-2.0.5-py2.7-linux-x86_64.egg/spotify/session.py", line 376, in get_link
    return spotify.Link(self, uri=uri)
  File "/usr/local/lib/python2.7/dist-packages/pyspotify-2.0.5-py2.7-linux-x86_64.egg/spotify/link.py", line 73, in __init__
    'Failed to get link from Spotify URI: %r' % uri)
ValueError: Failed to get link from Spotify URI: 'uri'

```

**Comment by jrnewell** [comment]
Yeah I get 'Track is not available, skipping...' and can't find it on the Spotify app.  I think it may be a spotify copyright / availablity / regional thing... Sorry!  Just to be sure, try doing a different track id just to check that the script works for you.

**Comment by TeNNoX** [comment]
**TL;DR** Omit the word 'uri' in your command.

I think the mistake is pretty simple here.
The format is
`spotify-ripper <OPTIONS> <URI>`

And because you passed 'uri' as uri, the error saying that the URI 'uri' is not valid is perfectly valid. (Sorry for the confusing wording :P)

Correct command would be:
`spotify-ripper -u andlessa -p XXX spotify:track:26xjIZq1zOMza3mfemA4Q8`
